pvlib/iam.py

```diff
@@ -940,3 +942,365 @@ def schlick_diffuse(surface_tilt):
+        ASHRAE model to the physical model.
+
+        When ``source_name`` is ``'ashrae'`` and ``target_name`` is
+        ``'physical'``, if `fix_n` is ``True``,
+        :py:func:`iam.convert` will fix ``n`` so that the returned physical
+        model has the same x-intercept as the inputted ASHRAE model.
+        Fixing ``n`` like this improves the fit of the conversion, but often
+        returns unrealistic values for the parameters of the physical model.
+        If more physically meaningful parameters are wanted,
+        set `fix_n` to False.
+
+    xtol : float, optional
+        Passed to scipy.optimize.minimize.
+
+    Returns
+    -------
+    dict
+        Parameters for the target model.
+
+            If target model is ``'ashrae'``, the dictionary will contain
+            the key ``'b'``.
+
+            If target model is ``'martin_ruiz'``, the dictionary will
+            contain the key ``'a_r'``.
+
+            If target model is ``'physical'``, the dictionary will
+            contain the keys ``'n'``, ``'K'``, and ``'L'``.
+
+    References
+    ----------
+    .. [1] TODO
+
+    See Also
+    --------
+    pvlib.iam.fit
+    pvlib.iam.ashrae
+    pvlib.iam.martin_ruiz
+    pvlib.iam.physical
+    """
+    if _min_scipy():
+
+        source = _get_model(source_name)
+        target = _get_model(target_name)
+
+        # if no options were passed in, we will use the default arguments
+        if weight is None:
+            weight = _sin_weight
+
+        aoi = np.linspace(0, 90, 100)
+        _check_params(source_name, source_params)
+        source_iam = source(aoi, **source_params)
+
+        if target_name == "physical":
+            # we can do some special set-up to improve the fit when the
+            # target model is physical
+            if source_name == "ashrae":
+                residual_function, guess, bounds = \
+                    _ashrae_to_physical(aoi, source_iam, weight, fix_n,
+                                        source_params['b'])
+            elif source_name == "martin_ruiz":
+                residual_function, guess, bounds = \
+                    _martin_ruiz_to_physical(aoi, source_iam, weight,
+                                             source_params['a_r'])
+
+        else:
+            # otherwise, target model is ashrae or martin_ruiz, and scipy
+            # does fine without any special set-up
+            bounds = [(1e-04, 1)]
+            guess = [1e-03]
+
+            def residual_function(target_param):
+                return _residual(aoi, source_iam, target, target_param, weight)
+
+        optimize_result = _minimize(residual_function, guess, bounds,
+                                    xtol=xtol)
+
+        return _process_return(target_name, optimize_result)
+
+    else:
+        return {}
+
+
+def fit(measured_aoi, measured_iam, target_name, weight=None, xtol=None):
+    """
+    Finds parameters for target model that best fit the
+    measured data.
```

Comment:
I suggest renaming ``target_name`` to ``model_name`` (as the fit function doesn't have a source model then it seems silly calling it a target model)